### PR TITLE
fivetran-destination: Start separate `COPY FROM` sinks for each file

### PR DESCRIPTION
**Pull request by ParkMyCar**
This PR updates the Fivetran Destination to change how it copies data into temporary tables within Materialize.

### Motivation

* Fixes an unspecified bug

Previously we would copy all CSV files via a single `COPY ... FROM` statement. This was sub-optimal because Materialize has a maximum buffer size when copying files. For example, if the maximum buffer size is 100MiB, you should be able to copy 2 80MiB files, you just need to use an individual `COPY ... FROM` for each. We were not doing this though so we would fail when trying to sync a large enough volume of data.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Improves the amount of data that can be synced with the Fivetran Destination

### CODE REVIEW

File: src/fivetran-destination/src/destination/dml.rs
Comment (jkosh44) [inline src/fivetran-destination/src/destination/dml.rs:1]: We still might fail though if any individual file is larger than the buffer?

**Review by jkosh44 — APPROVED** [review]
LGTM

File: src/fivetran-destination/src/destination/dml.rs
Comment (ParkMyCar) [reply src/fivetran-destination/src/destination/dml.rs:1]: We will, let me add a test for that

Comment (ParkMyCar) [reply src/fivetran-destination/src/destination/dml.rs:1]: Yes, let me add a test for that

**Pull request merged**